vllm/utils.py

```diff
@@ -338,7 +338,30 @@ def create_kv_caches_with_random(
     return key_caches, value_caches
 
 
-class measure_cuda_memory:
+_PIN_MEMORY_WARNING_PRINTED = False
+
+
+def pin_memory_available() -> bool:
+
+    def print_warning_once(msg: str) -> None:
+        global _PIN_MEMORY_WARNING_PRINTED
+        if not _PIN_MEMORY_WARNING_PRINTED:
+            _PIN_MEMORY_WARNING_PRINTED = True
+            logger.warning(msg)
+
+    if in_wsl():
+        # Pinning memory in WSL is not supported.
+        # https://docs.nvidia.com/cuda/wsl-user-guide/index.html#known-limitations-for-linux-cuda-applications
+        print_warning_once("Using 'pin_memory=False' as WSL is detected. "
+                           "This may slow down the performance.")
+        return False
+    elif is_neuron():
+        print_warning_once("Pin memory is not supported on Neuron.")
+        return False
+    return True
+
+
+class CudaMemoryMeasurer:
```

Comment:
nit: What about `CudaMemoryProfiler` or `CudaMemoryTracker` or something like that?